static-build/test/static-build/exports.test.lua

```diff
@@ -127,10 +127,48 @@ local check_symbols = {
     'HMAC_Update',
     'HMAC_Final',
 
+
+    --[[ ZSTD public API was grabbed using
+        gcc -DZSTDLIB_VISIBILITY=GREPAPI \
+            third_party/zstd/lib/zstd.h -E | grep '^GREPAPI'
+    ]]
```

Comment:
Those symbols does not exported explicitly. Maybe they should? Should the list be dependent on zstd version or be restricted to a certain common subset?

BTW, where we need those symbols?

We should not forget about `-DENABLE_BUNDLED_ZSTD=OFF` (not in the static build test, but at whole).

Reply:
> Those symbols does not exported explicitly. Maybe they should?

Unless the test fails I don't see a strong necessity.

> We should not forget about -DENABLE_BUNDLED_ZSTD=OFF (not in the static build test, but at whole).

It's a static build test and it implies `-DENABLE_BUNDLED_ZSTD=ON`

> Should the list be dependent on zstd version or be restricted to a certain common subset?

This list doesn't participate in the build process, but in testing only. The version of zstd is also pinned. And I already told @avtikhon that it should be updated in #5716. 

> BTW, where we need those symbols?

I'll answer later. Or maybe @HustonMmmavr helps me.

Reply:
Most of these symbols aren't used. Some of them is used in `src/box/xlog.c` and `src/box/vy_run.c`. I can't remember any rock that uses it via FFI though. But I'm sure that since it exists, the whole public API should be covered with a test.

Reply:
Usages inside tarantool are irrelevant: if such symbol will gone, we'll get linking error. They're not used inside tarantool via FFI. They don't exported, so external modules should not use them — it is not public API. And, from your words, it seems that they actually don't use them.

If we want to include those symbols into the public API, we should do several things:

1. See demand.
2. Add them to the export list.
3. File issue regarding the documentation team.